installers/rpm-code/codes/srw.sh

```diff
@@ -33,43 +33,51 @@ srw_main() {
-index 9e052df..0ab17fb 100644
---- a/env/work/srw_python/srwlib.py
-+++ b/env/work/srw_python/srwlib.py
-@@ -10267,6 +10267,8 @@ def srwl_wfr_emit_prop_multi_e(_e_beam, _mag, _mesh, _sr_meth, _sr_rel_prec, _n_
-                         #srwl_uti_save_intens_ascii(resStokes2.to_deg_coh(), meshRes2, fpdc2, _n_stokes = 1, _arLabels = resLabelsToSaveDC, _arUnits = resUnitsToSaveDC, _mutual = doMutual, _cmplx = 0) #OC12072019 # Deg. of Coh. Cut vs Y
-                         #srwl_uti_save_intens_ascii(resStokes2.to_deg_coh(), meshRes2, fpdc2, _n_stokes = 1, _arLabels = resLabelsToSaveDC, _arUnits = resUnitsToSaveDC, _mutual = 2, _cmplx = 0) #OC16072019 # Deg. of Coh. Cut vs Y
-                         srwl_uti_save_intens(resStokes2.to_deg_coh(), meshRes2, fpdc2, _n_stokes = 1, _arLabels = resLabelsToSaveDC, _arUnits = resUnitsToSaveDC, _mutual = 2, _cmplx = 0, _form = _file_form) #OC17072021 # Deg. of Coh. Cut vs Y
-+                elif _char == 61 and i + 1 < nRecv:
-+                    pass
-                 else:
-                     #srwl_uti_save_intens_ascii(resStokes.arS, meshRes, file_path1, numComp, _arLabels = resLabelsToSave, _arUnits = resUnitsToSave, _mutual = doMutual, _cmplx = (1 if doMutual else 0)) #OC30052017
-                     #srwl_uti_save_intens_ascii(resStokes.arS, meshRes, fp1, numComp, _arLabels = resLabelsToSave, _arUnits = resUnitsToSave, _mutual = doMutual, _cmplx = (1 if doMutual else 0)) #OC14082018
+srw_srwpy_backwards_compatible() {
+    declare -a old_modules=(
+        srwl_bl
+        srwl_uti_brightness
+        srwl_uti_cryst
+        srwl_uti_mag
+        srwl_uti_smp
+        srwl_uti_smp_rnd_obj2d
+        srwl_uti_smp_rnd_obj3d
+        srwl_uti_src
+        srwl_uti_und
+        srwlib
+        srwlpy
+        uti_io
+        uti_io_genesis
+        uti_mag
+        uti_math
+        uti_math_eigen
+        uti_parse
+        uti_plot
+        uti_plot_com
+        uti_plot_matplotlib
+    )
+
+    cd /home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/
+        cat > srwpy_import_warning.py <<EOF
+displayed = False
+
+def check(calling_module):
+    global displayed
+    if not displayed:
+        displayed = True
+        raise DeprecationWarning(f"This method of calling {calling_module} is deprecated. Please change to import from srwpy (ex: import {calling_module} -> from srwpy import {calling_module})")
 EOF
-}
+    for i in "${old_modules[@]}"
+    do
+        cat > $i.py <<EOF
+from srwpy.$i import *
+import srwpy_import_warning
 
-srw_python_patch_uti_math_eigen() {
-    patch --quiet env/work/srw_python/uti_math_eigen.py <<'EOF'
-@@ -29,7 +29,7 @@
- #from sklearn.decomposition import PCA as PCA #OC04082021 (moved to try-except)
- try:
-     from scipy.linalg import eigh as largest_eigh
--    from scipy.sparse.linalg.eigen.arpack import eigsh as largest_eigsh
-+    from scipy.sparse.linalg import eigsh as largest_eigsh
```

Comment:
I think this patch may be important too.

Reply:
patch not needed 994d56ec4eb7008c310ccbcf15d10cae03f86f38